### ISSUE TITLE
CP-JUMP1 dataset train/test split

### ISSUE THREAD

**Issue by Jeffery-MIC** [suinleelab/CellCLIP#4]
Hello. First, thank you for this fantastic work — I really appreciate the effort and clarity that went into the project. I’ve been exploring the CPJUMP1 evaluation benchmark, and while running it I ran into a few questions.

1. After downloading the dataset from the source, I want to run the script `src/tools/jumpcp_train_test_splitting.py` to acquire the testing set same as what you used. However, the script requires a file named `jumpcp_label.csv`. Would it be possible for you to provide the file or I should create it by myself?
2. Following the first question, if I have to create it by myself, how should I generate the `SAMPLE_KEY` column since I could not understand the original structure base on the parsing rules.
```python
# Line 28 from src/tools/jumpcp_train_test_splitting.py
label["batch"] = label["SAMPLE_KEY"].str.rsplit("-").str[0]
label["UNIQUE_SAMPLE_KEY"] = label["SAMPLE_KEY"].str.rsplit("-", n=1).str[0]
label = label.drop_duplicates(subset="UNIQUE_SAMPLE_KEY").reset_index(drop=True)
label["treatment"] = label["SAMPLE_KEY"].apply(lambda x: "-".join(x.split("-")[1:3]))
```

Thank you so much for your time reading this. If you need any detail while I was running, please feel free to request me to provide.

**Comment by q8888620002** [comment]
Hi @Jeffery-MIC 

I'm sorry for not getting back to you sooner; I missed this issue. I am glad to hear that you enjoy our work!

I couldn’t locate the original label file, but it follows the format below. You should be able to recreate your own using CPJUMP-1 with these columns:

```
ID,prompt,SAMPLE_KEY,Batch
BR00116991-P15-r16c15-05,"A549 cells treated with compound: olanzapine, SMILES: CN1CCN(CC1)C1=Nc2ccccc2Nc2sc(C)cc12",BR00116991-P15-r16c15-05,BR00116991
```

Hope this helps!
